Docs: debugging and persistent storage explanations (#325)
Unleash/unleash-edge: README.md

By default, Edge mode uses an in-memory cache to store the features it fetches from the upstream node. However, you may want to use a more persistent storage solution. For this purpose, Edge supports either Redis or a backup file, which you can configure by passing in either the `--redis-url` or `--backup_folder` command line argument, respectively. On start-up, Edge checks whether the persistent backup option is specified, in which case it uses it to populate its internal caches. This can be useful when your Unleash server is unreachable.

> Persistent storage is useful for high availability and resilience. If Edge using persistent storage is restarted, it will not have to synchronize with upstream before it is ready. Persistent storage is only read on start-up and valid until Edge is able to synchronize with upstream. This makes it so there is no need for purging cache. After Edge is synchronized with upstream it will periodically save its in-memory state to the persistent storage.

Reviewer: ```suggestion
Persistent storage is useful for high availability and resilience. If an Edge instance using persistent storage is restarted, it will not have to synchronize with upstream before it is ready. The persistent storage is only read on startup and will only be used until Edge is able to synchronize with upstream. Because of this, there is no need to purge the cache. After Edge has synchronized with an upstream, it will periodically save its in-memory state to the persistent storage.
```

> Multiple Edge nodes can share the same Redis instance or backup folder. Failure to read from persistent storage will not prevent Edge from starting up. It will delay it until Edge is able to connect with upstream.

Reviewer: What is "it" in this last sentence here?

Startup? being ready? Can we clarify that?

> Multiple Edge nodes can share the same Redis instance or backup folder. Failure to read from persistent storage will not prevent Edge from starting up. It will delay it until Edge is able to connect with upstream.

Author: ```suggestion
Multiple Edge nodes can share the same Redis instance or backup folder. Failure to read from persistent storage will not prevent Edge from starting up. In this case an Edge instance will be ready only after it is able to connect with upstream.
```

- Backend SDKs can connect to Unleash Edge without turning on experimental feature flags.
- Unleash Edge is smart enough to dynamically resolve the tokens you use to connect to it against the upstream Unleash instance. This means you don't have to worry about knowing in advance what tokens your SDKs use - if you want to swap out the Unleash token your SDK uses, this can be done without ever restarting or worrying about Unleash Edge. Unleash Edge will only collect and cache data for the environments and projects you use.

## Debugging

> You can adjust environment variable `RUST_LOG` to see more verbose log output, for example:

Author: I'll add a link to https://docs.rs/env_logger/0.10.0/env_logger/#enabling-logging

> You can adjust environment variable `RUST_LOG` to see more verbose log output, for example:

Author: ```suggestion
You can adjust environment variable `RUST_LOG` to see more verbose log output, for example raising default log level from `error` to `warning` and to `debug` for logs originating directly from Edge (and not other crates) can be done with the following value:
```

```sh

> RUST_LOG="warn,unleash-edge=debug" ./unleash-edge #<command>

Author: How about
> You can adjust environment variable `RUST_LOG` to see more verbose log output. For example raising default log level from `error` to `warning`, and setting it to `debug` for logs originating directly from Edge (not it's dependencies) can be done with the following value:

?

```

> 

Author: ```suggestion

See more about available log levels at https://docs.rs/env_logger/latest/env_logger/#enabling-logging

```

- Backend SDKs can connect to Unleash Edge without turning on experimental feature flags.
- Unleash Edge is smart enough to dynamically resolve the tokens you use to connect to it against the upstream Unleash instance. This means you don't have to worry about knowing in advance what tokens your SDKs use - if you want to swap out the Unleash token your SDK uses, this can be done without ever restarting or worrying about Unleash Edge. Unleash Edge will only collect and cache data for the environments and projects you use.

## Debugging

> You can adjust environment variable `RUST_LOG` to see more verbose log output. For example raising default log level from `error` to `warning` and to `debug` for logs originating directly from Edge (not it's dependencies) can be done with the following value:

Reviewer 2: Reads a bit weird to me. My suggestion:
```suggestion
You can adjust the `RUST_LOG` environment variable to see more verbose log output. For example, in order to get logs originating directly from Edge but not its dependencies, you can raise the default log level from `error` to `warning` and set Edge to `debug`, like this:
```
